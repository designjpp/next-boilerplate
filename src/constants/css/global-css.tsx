// const bodyCSS = global`
//   body {
//     @apply bg-white;
//   }

// import { Global } from '@emotion/react';
// import xw from 'xwind';

//   :global(#__next) {
//     @apply overflow-hidden;
//     @apply min-h-screen max-w-screen;
//     @apply relative;
//     @apply m-0 p-0;
//   }
// `;

// const safeAreaPaddingForNotchs = global`
//   body {
//     padding: env(safe-area-inset-top) env(safe-area-inset-right)
//       env(safe-area-inset-bottom) env(safe-area-inset-left);
//   }
// `;

// const fontOvverridesCSS = global`
//   h1, h2 {
//     font-family: 'Roboto', sans-serif;
//     @apply font-normal text-36px leading-42px uppercase;
//     @apply text-primary-default;
//     @apply max-w-640px;

//     letter-spacing: 0.48em;
//   }

//   h3 {
//     font-family: 'Roboto', sans-serif;
//     @apply font-normal text-24px leading-28px;
//     @apply text-primary-default;
//     @apply max-w-640px;
//   }

//   p,
//   span,
//   button,
//   a,
//   li,
//   h4,
//   h5,
//   h6 {
//     font-family: 'Roboto', sans-serif;
//     @apply font-normal text-14px leading-24px ;
//     @apply text-primary-default text-center;
//     @apply max-w-640px;
//   }
// `;

export const GlobalCSSList = () => (
	<>
		{/* <Global
			//tailwind base styles + keyframes + ring and shadow classes variables  ... to global styles
			styles={xw`XWIND_BASE XWIND_GLOBAL`}
		/> */}
	</>
);
