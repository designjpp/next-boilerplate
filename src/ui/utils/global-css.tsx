import { global } from 'styled-jsx/css';

export const GlobalCSS = () => (
	<>
		<style jsx global>
			{bodyCSS}
		</style>
		<style jsx global>
			{safeAreaPaddingForNotchs}
		</style>
		<style jsx global>
			{fontOvverridesCSS}
		</style>
	</>
);

const bodyCSS = global`
  body {
    @apply bg-dark;
  }

  :global(#__next) {
    @apply overflow-hidden;
    @apply min-h-screen max-w-screen;
    @apply relative flex flex-col items-center justify-center;
    @apply m-0;
  }
`;

const safeAreaPaddingForNotchs = global`
  body {
    padding: env(safe-area-inset-top) env(safe-area-inset-right)
      env(safe-area-inset-bottom) env(safe-area-inset-left);
  }
`;

const fontOvverridesCSS = global`
  h1, h2 {
    font-family: 'Roboto', sans-serif;
    @apply font-normal text-36px leading-42px uppercase;
    @apply text-primary-default text-center;
    @apply max-w-640px;

    letter-spacing: 0.48em;
  }

  h3 {
    font-family: 'Roboto', sans-serif;
    @apply font-normal text-24px leading-28px;
    @apply text-primary-default text-center;
    @apply max-w-640px;
  }

  p,
  span,
  button,
  a,
  li,
  h4,
  h5,
  h6 {
    font-family: 'Roboto', sans-serif;
    @apply font-normal text-14px leading-24px ;
    @apply text-primary-default text-center;
    @apply max-w-640px;
  }
`;
