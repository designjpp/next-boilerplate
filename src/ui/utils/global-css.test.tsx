import { create } from 'react-test-renderer';

import { GlobalCSS } from './global-css';

describe('<GlobalCSS>', () => {
	it("should matches it's snapshots", () => {
		const fontLinks = create(<GlobalCSS />).toJSON();
		expect(fontLinks).toMatchSnapshot();
	});
});
