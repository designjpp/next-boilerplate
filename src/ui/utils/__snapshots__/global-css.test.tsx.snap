// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalCSS> should matches it's snapshots 1`] = `null`;
